modules/gui/warp-list.lua

```diff
@@ -439,9 +439,10 @@ end)
 Gui.left_toolbar_button('item/'..config.default_icon,{'warp-list.main-tooltip',config.standard_proximity_radius},warp_list_container, function(player)
     return Roles.player_allowed(player,'gui/warp-list')
 end)
-:on_click(function(player,_,_)
-    local visible_state = Gui.toggle_left_element(player, warp_list_container)
-    keep_gui_open[player.name] = visible_state
+:on_custom_event(Gui.events.on_visibility_changed_by_click, function(player,_,event)
+    local state = event.state -- true if visible, false if invisible
+    -- Set gui keep open state for player that clicked the button
+    keep_gui_open[player.name] = state
```

Comment:
Could maybe have been one line plus comment with `keep_gui_open[player.name] = event.state`
May also be better to have it be `visible_state` but these are non blocking.

Reply:
https://github.com/Cooldude2606/scenario/pull/3/commits/8e9325fc0cf96dac9825a404796ed3d4283f1da6